v2/pkg/genruntime/core/errors.go

```diff
@@ -187,6 +187,40 @@ func (e *ConfigMapNotFound) Format(s fmt.State, verb rune) {
 	format(e, s, verb)
 }
 
+// SubscriptionMismatch error is used when a child resource and parent resource subscription don't match
+type SubscriptionMismatch struct {
+	cause error
+}
+
+func NewSubscriptionMismatchError(cause error) *SubscriptionMismatch {
+	return &SubscriptionMismatch{
+		cause: cause,
+	}
+}
+
+var _ error = &SubscriptionMismatch{}
+var _ causer = &SubscriptionMismatch{}
+
+func (e *SubscriptionMismatch) Error() string {
+	return e.cause.Error()
+}
```

Comment:
Does `SubscriptionMismatch` need a custom error message as well?

Reply:
I've changed how this is implemented so that it makes more sense. The error format is now part of this error type rather than just hardcoded in the file that called `NewSubscriptionMismatchError`